### PR TITLE
adding initial version of CODEOWNERS

### PR DESCRIPTION
**Pull request by philbucher**
see https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners
"You can use a CODEOWNERS file to define individuals or teams that are responsible for code in a repository."

=> I.e. if sth in a file that has a codeowner is modified, the owner is requested for review. This way always the right ppl will automatically be asked for review.

We can also enable that the codeoweners have to approve if their files are changed. Since we are a large repo with many ppl working on it I think this would good. We can discuss it, probably enable it in the future
image

I took inspiration in what the [Trilinos-Repo is doing](https://github.com/trilinos/Trilinos/blob/master/.github/CODEOWNERS)

@KratosMultiphysics/technical-committee this is what I told you about some time ago

Note that so far I have only done this for the core, but the same can/should be applied to the Apps one we have a bit experience with it.

### CODE REVIEW

**Comment by roigcarlo** [comment]
Like the idea 👍 👍 👍

**Comment by loumalouomega** [comment]
Nice

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee we decided to merge it as the first step toward getting better notification about the changes in the core. At this moment we don't activate the require reviews from the code owners to not making a drastic change in our approval process until we fill it would not make a bottleneck in the developments.

**Review by pooyan-dadvand — APPROVED** [review]
👍

**Comment by philbucher** [comment]
for the record, this seems to be working as expected :)

**Pull request merged**
